packages/framework/src/Framework/Views/Components/RelatedPublicationsComponent.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Hyde\Framework\Views\Components;
+
+use Hyde\Hyde;
+use Hyde\Publications\Models\PublicationPage;
+use Hyde\Publications\PublicationService;
+use Illuminate\Contracts\View\Factory;
+use Illuminate\Contracts\View\View;
+use Illuminate\Support\Collection;
+use Illuminate\View\Component;
+
+class RelatedPublicationsComponent extends Component
```

Comment:
This class is adding a considerable amount of complexity. Can you try adding some tests for it?

Reply:
Yes. I think the way to go is to refactor it into a few dedicated sub-methods which can then be tested for correctness per method and then test the entire thing with some fake data. I'll take a look at adding tests later this week.

Reply:
I think that sounds great.